test/Elastic.Apm.Tests/DistributedTracingTests.cs

```diff
@@ -0,0 +1,80 @@
+using System.Linq;
+using Elastic.Apm.DistributedTracing;
+using Elastic.Apm.Logging;
+using Elastic.Apm.Tests.Mocks;
+using FluentAssertions;
+using Xunit;
+
+namespace Elastic.Apm.Tests
+{
+	public class DistributedTracingTests
+	{
+
+		[Fact]
+		public void ParseValidTraceParentRecorded()
+		{
+			const string traceParent = "00-0af7651916cd43dd8448eb211c80319c-b7ad6b7169203331-01";
+
+			var(parent, traceId, isRecorded) = TraceParent.ParseTraceParentString(traceParent);
+
+			parent.Should().Be("0af7651916cd43dd8448eb211c80319c");
+			traceId.Should().Be("b7ad6b7169203331");
+			isRecorded.Should().BeTrue();
+		}
+
+		[Fact]
+		public void ParseValidTraceParentNotRecorded()
+		{
+			const string traceParent = "00-0af7651916cd43dd8448eb211c80319c-b7ad6b7169203331-00";
+
+			var(parent, traceId, isRecorded) = TraceParent.ParseTraceParentString(traceParent);
+
+			parent.Should().Be("0af7651916cd43dd8448eb211c80319c");
+			traceId.Should().Be("b7ad6b7169203331");
+			isRecorded.Should().BeFalse();
+		}
```

Comment:
It might be worth adding tests for flags value other than 0 and 1 to make sure is-recorded flag is extracted correctly even if other bits are not all zero. For example `0xC6` (`0b1100_0110`) and `0xE7` (`0b1110_0111`).

Reply:
Done, now we cover both.